### PR TITLE
COMMERCE-11367 Improve search in mini cart

### PR DESCRIPTION
**Pull request by smottal**
https://issues.liferay.com/browse/COMMERCE-11367

### CODE REVIEW

**Comment by smottal** [comment]
ci:test:sf

**Review by andrea-ale-sbarra — COMMENTED** [review]
@alessiorendina can you have a look at this too. It looks good to me overall, but I'm not sure about variables and function names.

Thank you!

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartQuickAdd.js [hunk 1]
```diff
@@ -18,95 +18,132 @@ import ClayForm, {ClayInput} from '@clayui/form';
...
 
 	const {cartItems = [], channel} = cartState;
 	const accountId = cartState.accountId;
 	const channelId = channel.channel.id;
 
 	useEffect(() => {
-		const productsApiURL = new URL(
-			`${themeDisplay.getPathContext()}${CHANNEL_RESOURCE_ENDPOINT}/${channelId}/products?accountId=${accountId}&nestedFields=skus&pageSize=-1&skus.accountId=${accountId}`,
+		if (paginatorCurrentPage > 1) {
+			invokeSearchAPI(productsQuery, true);
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartQuickAdd.js:55]: let's call this function searchProducts

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartQuickAdd.js [hunk 2]
```diff
@@ -18,95 +18,132 @@ import ClayForm, {ClayInput} from '@clayui/form';
...
+						})}
+				</ClayDropDown.ItemList>
+			</InfiniteScroller>
+		);
+	};
+
+	const getProductsApiURL = (page, search) => {
+		const url = new URL(
+			`${themeDisplay.getPathContext()}${CHANNEL_RESOURCE_ENDPOINT}/${channelId}/products`,
 			themeDisplay.getPortalURL()
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartQuickAdd.js:122]: what do you think of getSearchProductsURL?

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartQuickAdd.js [hunk 3]
```diff
@@ -18,95 +18,132 @@ import ClayForm, {ClayInput} from '@clayui/form';
...
-					else {
-						formattedProducts.push({
-							...product,
-							chipLabel: skus[0].sku,
-							label: name,
-							value: skus[0].sku,
-						});
-					}
-				});
+	const handleOnChangeQueryString = (queryString) => {
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartQuickAdd.js:135]: handleProductQueryInput(productQuery)

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
Please review the name of the functions.

Thank you!

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartQuickAdd.js [hunk 4]
```diff
@@ -18,95 +18,132 @@ import ClayForm, {ClayInput} from '@clayui/form';
...
-
 export default function CartQuickAdd() {
 	const {cartState, setCartState} = useContext(MiniCartContext);
 
+	const keypressTimoutRef = useRef(null);
+	const paginatorItemLengthRef = useRef(0);
+	const paginatorLastPageRef = useRef(1);
+	const paginatorIsLoadingRef = useRef(false);
+	const paginatorTotalCountRef = useRef(0);
+	const requestAbortControllerRef = useRef(new AbortController());
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartQuickAdd.js:39]: Sort alphabetically

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartQuickAdd.js [hunk 5]
```diff
@@ -184,6 +209,88 @@ export default function CartQuickAdd() {
 		}
 	};
 
+	const handleProductQueryInput = (queryString) => {
```
[the remark this answers is not in the document]
Comment (smottal) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartQuickAdd.js]: Changed in productQueryString

**Comment by smottal** [comment]
ci:test:commerce-functional

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:commerce-functional

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:commerce-functional

**Comment by smottal** [comment]
ci:test:relevant

**Comment by 22LucaGiannotta22** [comment]
ci:test:relevant

**Comment by 22LucaGiannotta22** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:relevant

**Comment by 22LucaGiannotta22** [comment]
ci:forward

**Pull request closed without merging**